server/storage/wal/testing/waltesting.go

```diff
@@ -49,7 +49,9 @@ func NewTmpWAL(t testing.TB, reqs []etcdserverpb.InternalRaftRequest) (*wal.WAL,
 		if err != nil {
 			t.Fatalf("Failed to open WAL: %v", err)
 		}
-		_, state, _, err := w.ReadAll()
+
+		var state raftpb.HardState
```

Comment:
Creating new var `state` to reduce the line changes. `err` doesn't reuse in other goroutines. using `err` here might be more easy to review.